templates/master/00-master/openstack/files/set-hostname.yaml

```diff
@@ -0,0 +1,22 @@
+filesystem: "root"
+mode: 0755
+path: "/usr/local/bin/set-hostname"
+contents:
+  inline: |
+    #!/usr/libexec/platform-python
+
+    import json
+    import urllib.request
+
+    # The url is static.
+    # Read more at https://docs.openstack.org/nova/latest/user/metadata.html
+    response = urllib.request.urlopen(
+        'http://169.254.169.254/openstack/2017-02-22/meta_data.json'
+    )
+
+    data = response.read()
+
+    decoded_data = json.loads(data)
+
+    with open('/etc/hostname', 'w') as f:
```

Comment:
Please add this to https://github.com/coreos/afterburn/ instead.

Reply:
See e.g. https://github.com/coreos/afterburn/pull/199 which landed exactly this for Azure.  By contributing to afterburn, you're also contributing to Fedora CoreOS, and the functionality works early from the OS start.

Oh also, doing this from a static pod will run straight into https://github.com/openshift/origin/pull/22826